### PR TITLE
Updates to the permission API

### PR DESCRIPTION
**Pull request by hamishwillee**
This updates the docs for Permission API, which were written when this was first proposed and very out of date as to status.

The main changes are to make it clear that this is mainstream now - so more exhaustive lists of supported APIs and their permissions. I couldn't find where `accessibility-events` is used though, so that isn't listed.

I've also found the fact that Permissions-Policy and Permissions API are the same to be a bit confusing, especially as they share the same permission names. I now understand that they are separate, but support each other. 
So the Permissions Policy is _one of the things_ that might restrict access to a resource on a page (context). The Permissions API is a way of working out what the aggregate permissions are, including user permission, and so on.

@Elchi3 FYI, trying to find out where those permissions are defined has highlighted that it would be great to move on adding more "Security Concerns" (or whatever we called them) sections. Mostly the information exists, but it could be aggregated, and that would be a great thing to link to in the API overviews.

This came up as I was trying to find decent info for #27748

### CODE REVIEW

File: files/en-us/web/api/permissions_api/index.md [hunk 1]
```diff
@@ -2,31 +2,48 @@
...
+The **Permissions API** provides a consistent programmatic way to query the status of API permissions attributed to the current context. For example, the Permissions API can be used to determine if permission to access a particular API has been granted or denied, or requires specific user permission.
 
-> **Note:** This feature is available in [Web Workers](/en-US/docs/Web/API/Web_Workers_API) (although not current versions of Firefox, as [WorkerNavigator.permissions](/en-US/docs/Web/API/WorkerNavigator/permissions) is not implemented).
+Note that the permissions from this API effectively aggregates all security restrictions for the context, including any requirement for an API to be used in a secure context, [Permissions-Policy](/en-US/docs/Web/HTTP/Headers/Permissions-Policy) restrictions applied to the document, and user prompts.
+So, for example, if an API is restricted by permission policy, the returned permission would be `denied` and the user would not be prompted for access.
```
Comment (sideshowbarker) [inline files/en-us/web/api/permissions_api/index.md:16]:
```suggestion
Note that the permissions from this API effectively aggregate all security restrictions for the context, including any requirement for an API to be used in a secure context, [Permissions-Policy](/en-US/docs/Web/HTTP/Headers/Permissions-Policy) restrictions applied to the document, and user prompts.
So, for example, if an API is restricted by permissions policy, the returned permission would be `denied` and the user would not be prompted for access.
```

File: files/en-us/web/api/permissions_api/index.md [hunk 2]
```diff
@@ -2,31 +2,48 @@
...
+
+> **Note:** This feature is available in [Web Workers](/en-US/docs/Web/API/Web_Workers_API) on platforms that support the [WorkerNavigator.permissions](/en-US/docs/Web/API/WorkerNavigator/permissions#browser_support) property.
 
 ## Concepts and usage
 
-Historically different APIs handle their own permissions inconsistently — for example the [Notifications API](/en-US/docs/Web/API/Notifications_API) allows for explicit checking of permission status and requesting permission, whereas the [Geolocation API](/en-US/docs/Web/API/Geolocation) doesn't (which causes problems if the user denied the initial permission request). The Permissions API provides the tools to allow developers to implement a better user experience as far as permissions are concerned.
+Historically different APIs handle their own permissions inconsistently — for example the [Notifications API](/en-US/docs/Web/API/Notifications_API) provided its own methods for requesting permissions and checking permission status, whereas the [Geolocation API](/en-US/docs/Web/API/Geolocation) did not.
```
Comment (sideshowbarker) [inline files/en-us/web/api/permissions_api/index.md:22]:
```suggestion
Historically different APIs handle their own permissions inconsistently — for example the [Notifications API](/en-US/docs/Web/API/Notifications_API) provided its own methods for requesting permissions and checking permissions status, whereas the [Geolocation API](/en-US/docs/Web/API/Geolocation) did not.
```

File: files/en-us/web/api/permissions_api/index.md [hunk 3]
```diff
@@ -2,31 +2,48 @@
...
-- [Clipboard API](/en-US/docs/Web/API/Clipboard_API)
-- [Notifications API](/en-US/docs/Web/API/Notifications_API)
-- [Push API](/en-US/docs/Web/API/Push_API)
-- Web MIDI API
-
-More APIs will gain Permissions API support over time.
+### Permission-aware APIs
+
+Not all APIs' permission statuses can be queried using the Permissions API.
+A non-exhaustive list of permission-aware APIs include:
```
Comment (sideshowbarker) [inline files/en-us/web/api/permissions_api/index.md:32]:
```suggestion
A non-exhaustive list of permission-aware APIs includes:
```

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/api/permissions_api/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/api/permissions_api/index.md:16]: Doh. Thank you.

**Pull request merged**